data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/GeoIpProviderFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.amazon.dataprepper.plugins.prepper.geoip.provider;
+
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.plugins.prepper.geoip.GeoIpPrepperConfig;
+
+import java.util.Objects;
+
+public class GeoIpProviderFactory {
+    public GeoIpProvider createGeoIpProvider(final PluginSetting pluginSetting) {
+        Objects.requireNonNull(pluginSetting);
+        final String providerString = pluginSetting.getStringOrDefault(GeoIpPrepperConfig.DATA_SOURCE, null);
+        Objects.requireNonNull(providerString, String.format("Missing '%s' configuration value", GeoIpPrepperConfig.DATA_SOURCE));
+
+        final DataSource provider = DataSource.valueOf(providerString);
+        switch (provider) {
+            case MaxMindGeolite2CityDatabase:
+                return new MaxMindGeoIpProvider(pluginSetting.getStringOrDefault(GeoIpPrepperConfig.DATABASE_PATH, null));
```

Comment:
If the database path is null, we should blow up here instead of passing around null values. (I know it'll get caught by the constructor args check regardless). This way we can keep the responsibility of arg validation in the factory.